core/src/com/unciv/logic/map/mapgenerator/MapLandmassGenerator.kt

```diff
@@ -28,7 +28,7 @@ class MapLandmassGenerator(val ruleset: Ruleset, val randomness: MapGenerationRa
     companion object {
         // this is called from TileMap constructors as well
         internal fun getInitializationTerrain(ruleset: Ruleset, type: TerrainType) =
-            ruleset.terrains.values.firstOrNull { it.type == type }?.name
+            ruleset.terrains.values.firstOrNull { it.type == type && it.isNaturallyGenerated() }?.name
```

Comment:
No need to add a special function to check if has unique, just put it here as a one-liner